docs: add guidelines for radio (#2147)
digdir/designsystemet: packages/react/src/components/form/Radio/Radio.mdx

Bruk `error` på `Radio.Group` for å vise feilmelding.

> Her må vi bruke `Radio.Group`, fordi den aktiverer riktig stil og sørger for at innholdet har de riktige attributtene når det gjelder tilgjengelighet.

Reviewer: ```suggestion
Her må vi bruke `Radio.Group`, fordi den aktiverer riktig stil og sørger for at innholdet har de riktige attributtene for tilgjengelighet.
```

### Readonly

Felter med `readonly`-attributtet er med i tabrekkefølgen. Brukerne kan kopiere innholdet men ikke redigere det. Informasjon blir med når skjemaet sendes inn.

`Readonly`-felter kan være forvirrende for noen brukere. Ikke alle vil skjønne hvorfor de ikke får til å endre innholdet i feltet. Vi anbefaler derfor å unngå readonly så langt det lar seg gjøre.

<Canvas of={RadioGroupStories.ReadOnly} />

### Disabled

> Unngå deaktiverte tilstander om du kan. De har lav fargekontrast som er problematisk for noen brukere. `Disabled` kan ikke møte kontrastkravene, for da kan brukeren tro at elementet er aktivt, prøve å trykke på det, og ikke skjønne hvorfor det ikke går.

Reviewer: ```suggestion
Unngå deaktiverte tilstander om du kan. De har lav fargekontrast som er problematisk for noen brukere. `disabled` kan ikke møte kontrastkravene, for da kan brukeren tro at elementet er aktivt, prøve å trykke på det, og ikke skjønne hvorfor det ikke går.
```

<Canvas of={RadioGroupStories.Disabled} />

### Inline

> Bruk `inline` for å plassere alternativene på samme linje. Anbefales ikke dersom du har mer enn to alternativer og lengre ledetekster.

Reviewer: ```suggestion
Bruk `inline` for å plassere alternativene på samme linje. Anbefales kun dersom du har to alternativer og korte ledetekster.
```
Synst det gir meir meining å anbefale når det skal brukast.

<Canvas of={RadioGroupStories.Inline} />

## Retningslinjer

Vi bruker `radio` når vi skal gi brukerne mulighet til å velge kun ett alternativ.

Passer til å

- velge ett alternativ

<br />

Passer ikke til å

- navigere
- velge flere alternativer samtidig
- vise mer enn syv alternativer

<br />

For grupper med radioknapper, kan vi ha opptil syv valg. Da viser vi dem helst på hver sin linje vertikalt, så det blir lett å skanne. Hvis vi bare har to valg med korte ledetekster, kan vi vise dem ved siden av hverandre.

Trenger vi å gi brukerne mer enn syv valg, bør vi heller bruke [`combobox`](/docs/komponenter-combobox--docs)-komponenten. Hvis vi bare skal gi brukerne ett valg, kan en [`switch`](/docs/komponenter-switch--docs) eller [`checkbox`](/docs/komponenter-checkbox--docs) passe bedre.

> Tenk nøye over om du virkelig trenger å ha et forhåndsvalgt alternativ. Det kan ha motsatt effekt av det du ønsker, for eksempel at brukeren føler seg manipulert til å ta ett bestemt valg eller ikke klarer å ta et bevisst valg selv.

Reviewer: ```suggestion
Tenk nøye over om du virkelig trenger å ha et forhåndsvalgt alternativ. Det kan ha motsatt effekt av det du ønsker, for eksempel at brukeren føler seg manipulert til å ta ett bestemt valg, eller ikke klarer å ta et bevisst valg selv.
```

### Readonly

Felter med `readonly`-attributtet er med i tabrekkefølgen. Brukerne kan kopiere innholdet men ikke redigere det. Informasjon blir med når skjemaet sendes inn.

> `Readonly`-felter kan være forvirrende for noen brukere. Ikke alle vil skjønne hvorfor de ikke får til å endre innholdet i feltet. Vi anbefaler derfor å unngå readonly så langt det lar seg gjøre.

Reviewer: ```suggestion
`readonly`-felter kan være forvirrende for noen brukere. Ikke alle vil skjønne hvorfor de ikke får til å endre innholdet i feltet. Vi anbefaler derfor å unngå `readonly` så langt det lar seg gjøre.
```

> `Readonly`-felter kan være forvirrende for noen brukere. Ikke alle vil skjønne hvorfor de ikke får til å endre innholdet i feltet. Vi anbefaler derfor å unngå `readonly` så langt det lar seg gjøre.

Reviewer: ```suggestion
`readonly`-felter kan være forvirrende for noen brukere. Ikke alle vil skjønne hvorfor de ikke får til å endre innholdet i feltet. Vi anbefaler derfor å unngå `readonly` så langt det lar seg gjøre.
```

<Canvas of={RadioGroupStories.ReadOnly} />

### Disabled

Unngå deaktiverte tilstander om du kan. De har lav fargekontrast som er problematisk for noen brukere. `disabled` kan ikke møte kontrastkravene, for da kan brukeren tro at elementet er aktivt, prøve å trykke på det, og ikke skjønne hvorfor det ikke går.

<Canvas of={RadioGroupStories.Disabled} />

### Inline

> Bruk `inline` for å plassere alternativene på samme linje. Anbefales kun dersom du har to alternativer og korte ledetekster (`labels`).

Reviewer: ```suggestion
Bruk `inline` for å plassere alternativene på samme linje. Anbefales kun dersom du har to alternativer og korte `label`s.
```

<Canvas of={RadioGroupStories.Inline} />

## Retningslinjer

Vi bruker `radio` når vi skal gi brukerne mulighet til å velge kun ett alternativ.

Passer til å

- velge ett alternativ

<br />

Passer ikke til å

- navigere
- velge flere alternativer samtidig
- vise mer enn syv alternativer

<br />

> Vi ha opptil syv alternativer i en gruppe. Da viser vi dem helst på hver sin linje vertikalt, så det blir lett å skanne. Hvis vi bare har to valg med korte ledetekster (`labels`), kan vi vise dem ved siden av hverandre.

Reviewer: ```suggestion
Vi kan ha opptil syv alternativer i en gruppe. Da viser vi dem helst på hver sin linje vertikalt, så det blir lett å skanne. Hvis vi bare har to valg med korte `label`s, kan vi vise dem ved siden av hverandre.
```

> Trenger vi å gi brukerne mer enn syv valg, bør vi heller bruke [`combobox`](/docs/komponenter-combobox--docs)-komponenten. Hvis vi bare skal gi brukerne ett valg, kan en [`switch`](/docs/komponenter-switch--docs) eller [`checkbox`](/docs/komponenter-checkbox--docs) passe bedre.

Reviewer: ```suggestion
Trenger vi å gi brukerne mer enn syv valg, bør vi heller bruke [`Combobox`](/docs/komponenter-combobox--docs). Hvis vi bare skal gi brukerne ett valg, kan en [`Switch`](/docs/komponenter-switch--docs) eller [`Checkbox`](/docs/komponenter-checkbox--docs) passe bedre.
```

Tenk nøye over om du virkelig trenger å ha et forhåndsvalgt alternativ. Det kan ha motsatt effekt av det du ønsker, for eksempel at brukeren føler seg manipulert til å ta ett bestemt valg, eller ikke klarer å ta et bevisst valg selv.

<br />

## Tekst i komponenten

En gruppe med alternativknapper bør alltid ha en `legend` og eventuelt en `description` dersom hjelpetekst er nødvendig.

> Radioknapper skal alltid ha en `label`. Den skal være så kort som mulig og hvert alternativ bør være formulert på samme måte. Radio-komponenten kan også ha hjelpetekst (`description`). Da kommer den rett under `label`.

Reviewer: ```suggestion
Radioknapper skal alltid ha en `label`. Den skal være så kort som mulig og hvert alternativ bør være formulert på samme måte. Radio-komponenten kan også ha `description`. Da kommer den rett under `label`.
```
